subprojects/language-java/src/main/java/org/gradle/language/java/plugins/JavaLanguagePlugin.java

```diff
@@ -77,10 +78,12 @@
  * @since 3.4
  */
 @Incubating
+@Deprecated
 public class JavaLanguagePlugin implements Plugin<Project> {
 
     @Override
     public void apply(Project project) {
+        SingleMessageLogger.nagUserOfReplacedPlugin("java-lang", "java-library");
```

Comment:
I think we should tweak the message. The `java-library` plugin isn't always the replacement.

I would add a section to the 'upgrading' document to describe what's going on and what to do about it, and then link to that from these deprecation nags and not suggest a specific replacement here.